packages/replay/src/coreHandlers/handleGlobalEvent.ts

```diff
@@ -0,0 +1,74 @@
+import { Event } from '@sentry/types';
+
+import { REPLAY_EVENT_NAME, UNABLE_TO_SEND_REPLAY } from '../constants';
+import type { ReplayContainer } from '../replay';
+import { addInternalBreadcrumb } from '../util/addInternalBreadcrumb';
+
+/**
+ * Returns a listener to be added to `addGlobalEventProcessor(listener)`.
+ */
+export function handleGlobalEventListener(replay: ReplayContainer): (event: Event) => Event {
+  return (event: Event) => {
+    // Do not apply replayId to the root event
+    if (
+      // @ts-ignore new event type
+      event.type === REPLAY_EVENT_NAME
+    ) {
+      // Replays have separate set of breadcrumbs, do not include breadcrumbs
+      // from core SDK
+      delete event.breadcrumbs;
+      return event;
+    }
+
+    // Only tag transactions with replayId if not waiting for an error
+    // @ts-ignore private
+    if (event.type !== 'transaction' || !replay._waitForError) {
+      event.tags = { ...event.tags, replayId: replay.session?.id };
+    }
+
+    // Collect traceIds in _context regardless of `_waitForError` - if it's true,
+    // _context gets cleared on every checkout
+    if (event.type === 'transaction') {
+      replay.getContext().traceIds.add(String(event.contexts?.trace?.trace_id || ''));
```

Comment:
l: we can try catch this and remove the need for optional chaining.

Reply:
I added a proper check here, as actually it is anyhow not ideal to add an empty trace id here, I guess.